unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -42,29 +38,39 @@ public RPCSocialApiBridge(IRPC rpc)
         public async UniTaskVoid InitializeClient(CancellationToken cancellationToken = default)
         {
             // start listening to streams
+            await UniTask.WhenAny(this.ListenToFriendEvents(cancellationToken));
         }
 
         public async UniTask<FriendshipInitializationMessage> InitializeFriendshipsInformation(CancellationToken cancellationToken = default)
         {
             cancellationToken.ThrowIfCancellationRequested();
+            var ownUserProfile = UserProfile.GetOwnUserProfile();
 
-            var friends = await rpc.Social().GetFriends(new Decentraland.Social.Friendships.Empty());
+            await UniTask.WaitUntil(() => ownUserProfile.userId != null);
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            var requestEvents = await rpc.Social().GetRequestEvents(new Decentraland.Social.Friendships.Empty());
+            var friendsStream = rpc.Social().GetFriends(new Empty());
+
+            await foreach (var friends in friendsStream)
```

Comment:
I think you can use `friendsStream.WithCancellation(cancellationToken)` here to support cancellation.